src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+    private final String value;
+    private boolean isPrivate;
+    
+    public Contact(String contact, boolean isPrivate) {
+        this.isPrivate = isPrivate;
+        contact = contact.trim();
+        this.value = contact;
+    }
```

Comment:
`MESSAGE_ADDRESS_CONSTRAINTS` and `ADDRESS_VALIDATION_REGEX` could have been passed to the parent class' constructor such that `isValidAddress` can be inherited from the parent class as well.